Apps/Src/SendCarCAN.c

```diff
@@ -1,35 +1,89 @@
-#include "Tasks.h"
-#include "CANbus.h"
-#include "CAN_Queue.h"
 #include "common.h"
+#include "os_cfg_app.h"
+#include "CANbus.h"
+#include "Minions.h"
+#include "Contactors.h"
+#include "Pedals.h"
+#include "Tasks.h"
+#include "SendCarCAN.h"
+
+// fifo
+#define FIFO_TYPE CANDATA_t
+#define FIFO_SIZE 256
+#define FIFO_NAME SendCarCAN_Q
+#include "fifo_protected.h"
+
+static SendCarCAN_Q_t CANFifo;
+
+#define IO_STATE_TMR_DLY_MS 250u
+#define IO_STATE_TMR_DLY_TS ((IO_STATE_TMR_DLY_MS * OS_CFG_TMR_TASK_RATE_HZ) / (1000u))
+static OS_TMR IOStateTmr;
+
+static void putIOState(void *p_tmr, void *p_arg);
 
 /**
- * @file SendCarCAN.c
- * @brief Implements the SendCarCAN Task
- * 
- * Resends the feedback from the motor controller
- * 
- */
+ * @brief Initialize SendCarCAN
+*/
+void SendCarCAN_Init(){
+    OS_ERR err;
+    
+    SendCarCAN_Q_renew(&CANFifo);
+    
+    // Set up timer to put car data message every IO_STATE_TMR_DLY_MS ms
+    OSTmrCreate(&IOStateTmr,
+                "IO State CAN Message Timer",
+                0,
+                IO_STATE_TMR_DLY_TS,
+                OS_OPT_TMR_PERIODIC,
+                putIOState,
+                NULL,
+                &err
+                );
+}
```

Comment:
Why is this being exported and called elsewhere instead of inside the Task's initialization sequence? Doesn't really make sense for the application not to initialize it's core data structure.

Reply:
I thought we might need it considering that other tasks may be initialized before SendCarCAN and might try to put something in the queue before the semaphore & mutex are initialized. If we initialize all of our tasks before starting the OS though, it's not a problem and I can initialize directly in the task.